.github/workflows/pr-lint.yml

```diff
@@ -0,0 +1,42 @@
+name: "Lint PR"
+
+on:
+  pull_request_target:
+    types:
+      - opened
+      - edited
+      - synchronize
+
+jobs:
+  main:
+    runs-on: ubuntu-latest
+    steps:
+      # Please look up the latest version from
+      # https://github.com/amannn/action-semantic-pull-request/releases
+      - uses: amannn/action-semantic-pull-request@v3.4.0
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          types: |
+            feat
+            fix
+            chore
+            refactor
+            revert
+          scopes: |
+            lib
+            cli
+            hcl2json
+            tests
+            examples
+            readme
+            docs
+            release
+            deps
+          # Configure that a scope must always be provided.
+          requireScope: true
+          # When using "Squash and merge" on a PR with only one commit, GitHub
+          # will suggest using that commit message instead of the PR title for the
+          # merge commit, and it's easy to commit this by mistake. Enable this option
+          # to also validate the commit message for one commit PRs.
```

Comment:
From past experience I would suggest we add a `subjectPatternError` to make it clear why this linter exists. I'd suggest something like

> The subject "{subject}" found in the pull request title "{title}" does not match the required format. The format is "type(scope): message", supported types are feat, fix, chore, refactor, revert. The supported scopes are lib, cli, hcl2json, tests, examples, readme, docs, release, deps. Please stick to the format so that our automatic release process picks your commit up and your code makes it into the next release.

Reply:
Yes, I like that 👍